pkg/operator/controller/ingress/controller_test.go

```diff
@@ -317,3 +317,60 @@ func TestTLSProfileSpecForIngressController(t *testing.T) {
 		}
 	}
 }
+
+func TestValidateHTTPHeaderBufferValues(t *testing.T) {
+	testCases := []struct {
+		description      string
+		httpHeaderBuffer operatorv1.IngressControllerHTTPHeaderBuffer
+		valid            bool
+	}{
+		{
+			description:      "empty httpHeaderBuffer",
+			httpHeaderBuffer: operatorv1.IngressControllerHTTPHeaderBuffer{},
+			valid:            true,
+		},
+		{
+			description: "valid, non-empty httpHeaderBuffer",
+			httpHeaderBuffer: operatorv1.IngressControllerHTTPHeaderBuffer{
+				HeaderBufferBytes:           32768,
+				HeaderBufferMaxRewriteBytes: 8192,
+			},
+			valid: true,
+		},
+		{
+			description: "invalid, non-empty httpHeaderBuffer",
+			httpHeaderBuffer: operatorv1.IngressControllerHTTPHeaderBuffer{
+				HeaderBufferBytes:           8192,
+				HeaderBufferMaxRewriteBytes: 32768,
+			},
+			valid: false,
+		},
+		{
+			description: "invalid httpHeaderBuffer, HeaderBufferMaxRewriteBytes empty",
+			httpHeaderBuffer: operatorv1.IngressControllerHTTPHeaderBuffer{
+				HeaderBufferBytes: 1,
+			},
+			valid: false,
+		},
+		{
+			description: "invalid httpHeaderBuffer, HeaderBufferBytes empty",
+			httpHeaderBuffer: operatorv1.IngressControllerHTTPHeaderBuffer{
+				HeaderBufferMaxRewriteBytes: 16384,
+			},
+			valid: true,
```

Comment:
The description says this is "invalid".

Reply:
whoops, needed to flip that and raise the value in the test. Fixed.